From 912bf10b639170e527a539ec80cdee840a370e89 Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Thu, 29 Mar 2018 19:58:45 +0300
Subject: [PATCH 135/706] enetc: enable MAC promisc mode, is required per BG
 123

It does default to promisc, but the driver resets it and it is required
to receive unicast.  We only filter traffic at later stages in HW.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
(cherry picked from commit dc6b4a665d7c76ca111b1cba0570a92a73dc909e)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_hw.h | 1 +
 drivers/net/ethernet/freescale/enetc/enetc_pf.c | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_hw.h b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
index 3df226016c9c..4a87f22a7fd4 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_hw.h
+++ b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
@@ -207,6 +207,7 @@ enum enetc_bdr_type {TX, RX};
 #define ENETC_PM0_CMD_CFG	0x08008
 #define ENETC_PM0_TX_EN		BIT(0)
 #define ENETC_PM0_RX_EN		BIT(1)
+#define ENETC_PM0_PROMISC	BIT(4)
 #define ENETC_PM0_CMD_XGLP	BIT(10)
 #define ENETC_PM0_CMD_TXP	BIT(11)
 
diff --git a/drivers/net/ethernet/freescale/enetc/enetc_pf.c b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
index eaef1573aad0..21817a94339d 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_pf.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
@@ -501,7 +501,8 @@ static void enetc_configure_port_mac(struct enetc_hw *hw)
 		      ENETC_SET_MAXFRM(ENETC_RX_MAXFRM_SIZE));
 
 	enetc_port_wr(hw, ENETC_PM0_CMD_CFG,
-		      ENETC_PM0_CMD_TXP	| ENETC_PM0_TX_EN | ENETC_PM0_RX_EN);
+		      ENETC_PM0_CMD_TXP	| ENETC_PM0_PROMISC |
+		      ENETC_PM0_TX_EN | ENETC_PM0_RX_EN);
 }
 
 static void enetc_configure_port(struct enetc_pf *pf)
-- 
2.17.1

